authors/apps/authentication/tests/test_user_registration.py

```diff
@@ -0,0 +1,249 @@
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_no_username_field(self):
+        """Test for user registration without a username field."""
+        no_username_field_data = {
+            "user": {
+                "email": "john@gmail.com",
+                "password": "John1234"
+            }
+        }
+        no_username_field_data_response = {"errors": {
+            "username": ["Username is required"]
+        }
+        }
+        response = self.client.post(self.url, no_username_field_data, format='json')
+        self.assertEqual(response.data, no_username_field_data_response)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_existing_username(self):
+        """Test for user registration with an existing username."""
+        username_data = {
+            "user": {
+                "username": "Mary",
+                "email": "mary@gmail.com",
+                "password": "John1234"
+            }
+        }
+        existing_username_data = {
+            "user": {
+                "username": "Mary",
+                "email": "maryg@gmail.com",
+                "password": "John1234"
+            }
+        }
+        existing_username_data_response = {"errors": {
+            "username": ["A user with this username already exists"]
+        }
+        }
+        self.client.post(self.url, username_data, format='json')
+        response = self.client.post(self.url, existing_username_data, format='json')
+        self.assertEqual(response.data, existing_username_data_response)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_blank_email(self):
+        """Test for user registration with a blank email."""
+        blank_email_data = {
+            "user": {
+                "username": "John",
+                "email": "",
+                "password": "John1234"
+            }
+        }
+        blank_email_data_response = {"errors": {
+            "email": ["Email field cannot be blank"]
+        }
+        }
+        response = self.client.post(self.url, blank_email_data, format='json')
+        self.assertEqual(response.data, blank_email_data_response)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_no_email_field(self):
+        """Test for user registration without an email field."""
+        no_email_field_data = {
+            "user": {
+                "username": "John",
+                "password": "John1234"
+            }
+        }
+        no_email_field_data_response = {"errors": {
+            "email": ["Email is required"]
+        }
+        }
+        response = self.client.post(self.url, no_email_field_data, format='json')
+        self.assertEqual(response.data, no_email_field_data_response)
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_invalid_email(self):
+
```

Comment:
@RachelleMaina, please be consistent and not leave a space before the docstrings as all your other docstrings do not have a space at the top.

Reply:
Will do this